gefserver/pier/pier.go

```diff
@@ -148,8 +149,9 @@ func (p *Pier) RunService(service db.Service, inputPID string) (db.Job, error) {
 	}
 
 	err := p.db.AddJob(job)
+	updService, err := p.db.GetService(service.ID) // we need an updated service with all metadata
```

Comment:
Either this is not needed, or RunService should take as input the service id. Check how RunService is used.